pandas/core/internals/blocks.py

```diff
@@ -878,6 +881,11 @@ def setitem(self, indexer, value):
             values[indexer] = value
             return self.make_block(Categorical(self.values, dtype=arr_value.dtype))
 
+        elif exact_match and is_ea_value:
```

Comment:
@jbrockmendel this entire routine needs some refactoring to avoid these types of patterns (but later).

Reply:
yep, ive got a PR on deck for once this goes in